### PR TITLE
change checks of ODO_SRC_PATH/src in tests and a temporary fix for project list flake

### PR DESCRIPTION
**Pull request by girishramnani**
**What kind of PR is this?**

/kind bug
**What does does this PR do / why we need it**:
Due to recent change in the s2i image for nodejs in openshift 4.3. The `/tmp/src` directory doesn't hold the code anymore because it gets moved. 
We know this happens for some images and we have a workaround in the odo-init-image. But our test for nodejs assumes that source is present in `/tmp/src`, due to which they are failing on openshift 4.3.

**Which issue(s) this PR fixes**:

Fixes https://github.com/openshift/odo/issues/2357

**How to test changes / Special notes to the reviewer**:
tests should pass on 4.3 cluster

### CODE REVIEW

**Comment by girishramnani** [comment]
so this issue https://github.com/openshift/odo/issues/2359 is failing on this PR

**Comment by girishramnani** [comment]
Debug mode seems to be failing

**Comment by dharmit** [comment]
I have a silly question. How do we know which tests need to be changed to use `nodejs:8` instead of `nodejs`? I mean, in spite of the latest commit pushed to address the test failures, there are still various places in the code where we're using `nodejs`, no? @girishramnani @kadel

**Comment by girishramnani** [comment]
> I have a silly question. How do we know which tests need to be changed to use `nodejs:8` instead of `nodejs`? I mean, in spite of the latest commit pushed to address the test failures, there are still various places in the code where we're using `nodejs`, no? @girishramnani @kadel

in the tests where we changed to `nodejs:8` the internals are being tested.

File: tests/integration/cmd_push_test.go [hunk 1]
```diff
@@ -197,72 +197,78 @@ var _ = Describe("odo push command tests", func() {
...
-				func(cmdOp string, err error) bool {
-					modifiedCatServerFile = cmdOp
-					return true
-				},
-			)
-
-			Expect(modifiedCatViewFile).NotTo(Equal(earlierCatViewFile))
-			Expect(modifiedCatServerFile).To(Equal(earlierCatServerFile))
-		})
+		// these tests are failing with stat output being different
```
Comment (kadel) [inline tests/integration/cmd_push_test.go]: Why is stat output different? We shouldn't comment out tests just because they are failing.
/hold

Comment (girishramnani) [reply tests/integration/cmd_push_test.go]: resolved @kadel

File: tests/integration/cmd_push_test.go [hunk 2]
```diff
@@ -197,9 +197,9 @@ var _ = Describe("odo push command tests", func() {
 			Expect(output).To(Not(ContainSubstring("No file changes detected, skipping build")))
 		})
 
-		It("should push only the modified files", func() {
+		FIt("should push only the modified files", func() {
```
Comment (kadel) [inline tests/integration/cmd_push_test.go]: Focused test.

Comment (girishramnani) [reply tests/integration/cmd_push_test.go]: resolved

File: tests/integration/cmd_debug_test.go [hunk 3]
```diff
@@ -34,10 +34,10 @@ var _ = Describe("odo debug command tests", func() {
 		os.Unsetenv("GLOBALODOCONFIG")
 	})
 
-	Context("odo debug on a nodejs component", func() {
+	Context("odo debug on a nodejs:8 component", func() {
```
Comment (kadel) [inline tests/integration/cmd_debug_test.go]: why was this changed to `8`?

Comment (girishramnani) [reply tests/integration/cmd_debug_test.go]: Because the latest nodejs image is not working with debug mode and hence also it's now shown as fully supported

Comment (girishramnani) [reply tests/integration/cmd_debug_test.go]: We need to resolve the debug mode issue for latest node in a separate PR

Comment (girishramnani) [reply tests/integration/cmd_debug_test.go]: Issue for the broken debug mode https://github.com/openshift/odo/issues/2359#issuecomment-550284845

Comment (girishramnani) [reply tests/integration/cmd_debug_test.go]: We also have an issue which mentions that we should run test across all the image versions. So we have recorded everything that has to be done after this fix

File: tests/integration/cmd_push_test.go [hunk 4]
```diff
@@ -55,27 +55,27 @@ var _ = Describe("odo push command tests", func() {
 		})
 
 		It("should be able to create a file, push, delete, then push again propagating the deletions", func() {
-			helper.CmdShouldPass("git", "clone", "https://github.com/openshift/ruby-ex", context+"/ruby-ex")
-			helper.CmdShouldPass("odo", "component", "create", "ruby", cmpName, "--project", project, "--context", context+"/ruby-ex", "--app", appName)
+			helper.CmdShouldPass("git", "clone", "https://github.com/openshift/nodejs-ex", context+"/nodejs-ex")
```
Comment (cdrage) [inline tests/integration/cmd_push_test.go]: Why are we changing it from ruby to nodejs? Is it because we don't "officially" support ruby (yet?)

Comment (girishramnani) [reply tests/integration/cmd_push_test.go]: Yes and some variables like `DEPLOYMENT_DIR` are not populated in ruby s2i images

**Comment by amitkrout** [comment]
Review comments are addressed. Thanks @girishramnani 
/approve
